### PR TITLE
examples: add error_handling example; move errors to error_details

### PR DESCRIPTION
**Pull request by dfawley**
RELEASE NOTES: none

### CODE REVIEW

File: examples/features/error_handling/client/main.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/*
+ *
+ * Copyright 2018 gRPC authors.
```
Comment (arvindbr8) [inline examples/features/error_handling/client/main.go:3]: Back to the future 🔁

File: examples/features/error_handling/server/main.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/*
+ *
+ * Copyright 2018 gRPC authors.
```
Comment (arvindbr8) [inline examples/features/error_handling/server/main.go:3]: Back to the future 2 🔁

File: examples/features/error_handling/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+This example demonstrates basic RPC error handling in gRPC.
+
+# Run the sample code
+
+Run the server, which returns an error if the RPC request's `Name` field is
+empty.
+
+```sh
+$ go run ./server/main.go
```
Comment (arvindbr8) [inline examples/features/error_handling/README.md:11]: I've seen that the shell command in other READMEs in our repo start with a `$`. But when we preview the markdown, there is usually a copy to clipboard option for these codeblocks; which copies the `$` prompt and needs further editing. 

what do you think? Should we start removing them from here on now? Or keep it as is (like the rest of them)

File: examples/features/error_handling/server/main.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+)
+
+var port = flag.Int("port", 50052, "port number")
+
+// server is used to implement helloworld.GreeterServer.
+type server struct {
+	pb.UnimplementedGreeterServer
+}
+
+// SayHello implements helloworld.GreeterServer
```
Comment (arvindbr8) [inline examples/features/error_handling/server/main.go:43]:
```suggestion
// SayHello implements helloworld.GreeterServer.
```

however, the period after a fieldname is weird for me too.

**Review by arvindbr8 — CHANGES_REQUESTED** [review]
thank you for doing this; else we would be losing to gRPC-Java in the great examples races. 

Very minor comments. PTAL

File: examples/features/error_handling/client/main.go (hunk 1 above)
Comment (dfawley) [reply examples/features/error_handling/client/main.go:3]: Pitfalls of `cp -R` I guess.

File: examples/features/error_handling/README.md (hunk 3 above)
Comment (dfawley) [reply examples/features/error_handling/README.md:11]: If we want to make a change, let's do it as a cleanup PR across all examples.  The problem with not using the `$` is more apparent when there is example output accompanying it.  I think this is minor enough that we don't need to care much either way, and we should aim to be consistent.

**Comment by dfawley** [comment]
Updated; PTAL

File: examples/features/error_handling/README.md (hunk 3 above)
Comment (arvindbr8) [reply examples/features/error_handling/README.md:11]: SGTM

**Review by arvindbr8 — APPROVED** [review]

**Pull request merged**